src/core/thread/network_data.cpp

```diff
@@ -519,36 +463,20 @@ bool NetworkData::ContainsService(uint8_t aServiceId, uint16_t aRloc16)
 void NetworkData::RemoveTemporaryData(uint8_t *aData, uint8_t &aDataLength)
 {
     NetworkDataTlv *cur = reinterpret_cast<NetworkDataTlv *>(aData);
-    NetworkDataTlv *end;
-    PrefixTlv *     prefix;
-    ServiceTlv *    service;
-    uint8_t         length;
-    uint8_t *       dst;
-    uint8_t *       src;
-
-    while (1)
-    {
-        end = reinterpret_cast<NetworkDataTlv *>(aData + aDataLength);
-
-        if (cur >= end)
-        {
-            break;
-        }
 
+    while (cur < reinterpret_cast<NetworkDataTlv *>(aData + aDataLength))
+    {
         switch (cur->GetType())
         {
         case NetworkDataTlv::kTypePrefix:
         {
-            prefix = static_cast<PrefixTlv *>(cur);
+            PrefixTlv *prefix = static_cast<PrefixTlv *>(cur);
```

Comment:
nit: use reference instead of pointer?

Reply:
Good idea.
In new push I added a new commit adding a templated version `FindTlv<PrefixTlv>` which directly returns a `PrefixTlv *`. This helps removes all the `static_cast`.